sdk/core/core/src/az_client_secret_credential.c

```diff
@@ -135,13 +147,50 @@ static AZ_NODISCARD az_result az_token_credential_get_token(
         auth_url));
 
     AZ_RETURN_IF_FAILED(az_http_request_builder_add_body(&hrb, auth_body));
-    AZ_RETURN_IF_FAILED(az_http_client_send_request_and_get_body(&hrb, &response_buf));
+
+    static az_http_pipeline pipeline = {
+    .policies = {
+      { .pfnc_process = no_op_policy, .data = NULL },
```

Comment:
why do we need no-op policies?

Reply:
Thanks, I'll make changes to add other policies.

Reply:
Done